services/ark-dashboard/ark-dashboard/__tests__/unit/components/cards/a2a-server-card.test.tsx

```diff
@@ -0,0 +1,214 @@
+import { describe, it, expect, vi } from 'vitest';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { A2AServerCard } from '@/components/cards/a2a-server-card';
+import type { A2AServer } from '@/lib/services/a2a-servers';
+
+vi.mock('@/lib/utils/icon-resolver', () => ({
+  getCustomIcon: vi.fn(() => () => <div>IconMock</div>),
+}));
+
+vi.mock('@/components/dialogs/confirmation-dialog', () => ({
+  ConfirmationDialog: vi.fn(({ open, title, onConfirm, confirmText }) =>
+    open ? (
+      <div data-testid="confirmation-dialog">
+        <div>{title}</div>
+        <button onClick={onConfirm}>{confirmText}</button>
+      </div>
+    ) : null
+  ),
+}));
+
+vi.mock('@/components/ui/availability-status-badge', () => ({
+  AvailabilityStatusBadge: vi.fn(({ status, eventsLink }) => (
+    <a href={eventsLink} data-testid="availability-badge">
+      Status: {status}
+    </a>
+  )),
+}));
+
+vi.mock('./base-card', () => ({
+  BaseCard: vi.fn(({ title, actions, footer }) => (
+    <div data-testid="base-card">
+      <div>{title}</div>
+      {actions.map((action: { label: string; onClick: () => void }, idx: number) => (
+        <button key={idx} onClick={action.onClick} aria-label={action.label}>
+          {action.label}
+        </button>
+      ))}
+      {footer}
+    </div>
+  )),
+}));
+
+describe('A2AServerCard', () => {
+  const mockA2AServer: A2AServer = {
+    id: 'test-id',
+    name: 'test-server',
+    namespace: 'default',
+    ready: true,
+    address: 'http://test.example.com',
+  };
+
+  it('should render server name and address', () => {
+    render(
+      <A2AServerCard
+        a2aServer={mockA2AServer}
+        namespace="default"
+      />
+    );
+
+    expect(screen.getByText('test-server')).toBeInTheDocument();
+    expect(screen.getByText(/http:\/\/test\.example\.com/)).toBeInTheDocument();
+  });
+
+  it('should render delete button when onDelete provided', () => {
+    const onDelete = vi.fn();
+    render(
+      <A2AServerCard
+        a2aServer={mockA2AServer}
+        namespace="default"
+        onDelete={onDelete}
+      />
+    );
+
+    expect(screen.getByLabelText('Delete a2a server')).toBeInTheDocument();
+  });
+
+  it('should not render delete button when onDelete not provided', () => {
+    render(
+      <A2AServerCard
+        a2aServer={mockA2AServer}
+        namespace="default"
+      />
+    );
+
+    expect(screen.queryByLabelText('Delete a2a server')).not.toBeInTheDocument();
+  });
+
+  it('should show confirmation dialog on delete click', () => {
+    const onDelete = vi.fn();
+    render(
+      <A2AServerCard
+        a2aServer={mockA2AServer}
+        namespace="default"
+        onDelete={onDelete}
+      />
+    );
+
+    fireEvent.click(screen.getByLabelText('Delete a2a server'));
```

Comment:
consider using the `userEvent` API instead of the lower level `fireEvent`

[here the differences between these two
](https://testing-library.com/docs/user-event/intro/#difference-to-fireevent)

```suggestion
    userEvent.click(screen.getByRole('button', { name: /delete/i }));
```

Reply:
done